### PR TITLE
Fixed url creating for feach features

### PR DESCRIPTION
**Pull request by VovaStelmashchuk**
- Fixed url creating, now url creating in case when host doesn't have dash.
- Add contributing guide line.

Simplified version of #10

### CODE REVIEW

**Comment by VovaStelmashchuk** [comment]
Fixes for cases when a developer provide a host with slash or without slash. Each case are correct, but library create incorrect request the host in case when url without slash.

**Comment by VovaStelmashchuk** [comment]
@stevenleadbeater Can you check error into `Test coverage`. I see some package cannot install, may be needs some changes into action source.

**Comment by stevenleadbeater** [comment]
@VovaStelmashchuk let me take a look at that for you

**Comment by VovaStelmashchuk** [comment]
A problem is a pipeline tries to install a package. I think it is a wrong way for CI. The pipeline  has depends on ubuntu package storage. 

My proposal don't use bash/shell executor into CI instead create Docker image with all tools which you need to the pipeline 
job. 
[Creating a Docker container action](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action)

Pluses:
- CI runs faster  
- No depends
- Developers can use the docker image for run ci jobs localy.

[Link to CI template how uses shell](https://github.com/stevenleadbeater/kotlin-kover-action/blob/main/action.yml)

**Comment by stevenleadbeater** [comment]
Yes you're right there. I've done docker containers for actions before. I'll refactor this one to use that style too. I had used the composite action instead originally as it's simpler to setup and less moving parts. I've got a docker quay account anyway so there's no problems converting it. Give me a bit and I'll fix it up this morning UST time 😉

**Comment by stevenleadbeater** [comment]
All fixed @VovaStelmashchuk , please push a whitespace change to ensure the builds run using the latest versions of the actions involved

**Comment by stevenleadbeater** [comment]
looking at this now, please stand by

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ jobs:
       with:
         coverage-threshold: 80
         repo-token: ${{ secrets.GITHUB_TOKEN }}
+        proxy-url: https://add-pr-comment-proxy-94idvmwyie-uc.a.run.app
```
Comment (stevenleadbeater) [inline .github/workflows/pull-request.yml:17]: There are warnings about this in the workflow output. Can you remove it please?

**Review by stevenleadbeater — CHANGES_REQUESTED** [review]

**Comment by stevenleadbeater** [comment]
> Ok, my suggest [stevenleadbeater/kotlin-kover-action#1](https://github.com/stevenleadbeater/kotlin-kover-action/pull/1)
> 
> I am not sure about solution at the moment :)

I had some similar issues going on on my Fork. It may be using an older version of the action. I resolved it by closing the pull request and opening a new one

**Comment by stevenleadbeater** [comment]
Run mshick/add-pr-comment@v1
  with:
    message: | Area  | Coverage(%) | Status |
  | ------------- | ------------- | ------------- |
  | **Line**  | 94  | :heavy_check_mark: |
  | **Instruction**  | 85  | :heavy_check_mark:  |
  | **Branch**  | 66  | :x:  |
  | **Method**  | 88  | :heavy_check_mark:  |
  | **Class**  | 89  | :heavy_check_mark:  |
  
    repo-token: ***
    repo-token-user-login: github-actions[bot]
    allow-repeats: false
  env:
    JAVA_HOME_11.0.14_x64: /opt/hostedtoolcache/jdk/11.0.14/x64
    JAVA_HOME: /opt/hostedtoolcache/jdk/11.0.14/x64
    JAVA_HOME_11_0_14_X64: /opt/hostedtoolcache/jdk/11.0.14/x64
Error: Resource not accessible by integration

**Comment by stevenleadbeater** [comment]
Found you:
Proxy for Fork-based PRs

GitHub limits GITHUB_TOKEN and other API access token permissions when creating a PR from a fork. This precludes adding comments when your PRs are coming from forks, which is the norm for open source projects. To work around this situation I've created a simple companion app you can deploy to Cloud Run or another host to proxy the create comment requests with a personal access token you provide.

**Comment by stevenleadbeater** [comment]
this needs adding in to the composite action. Please stand by while I test a fix :)

**Comment by stevenleadbeater** [comment]
Please add the:
        `proxy-url: https://add-pr-comment-proxy-94idvmwyie-uc.a.run.app`
back in, it is now supported in the action and should resolve the build failure

**Comment by stevenleadbeater** [comment]
This is missing the comments on coverage, that's a pain. It looks like it's related to tokens not being accessible to PRs from forks. It's a security risk as anyone can open a fork, if they then modify the workflow they can potentially exfiltrate the token and do all sorts of nasty things. I'm going to spend a bit of time on this and see what's possible. there's the proxy with access to write comments that you found but that requires an instance running in google cloud and seems prone to breaking and failure. I'll keep looking at this. I would like to find a solution that doesn't require proxies or manual steps and retains the security of the repository

**Comment by stevenleadbeater** [comment]
Looks like codecov is the way to go for this. I'll get everything updated and integrate it back in. For now, your coverage is:
  | message: | Area  | Coverage(%) | Status |
  | ------------- | ------------- | ------------- |
  | **Line**  | 94  | :heavy_check_mark: |
  | **Instruction**  | 85  | :heavy_check_mark:  |
  | **Branch**  | 66  | :x:  |
  | **Method**  | 88  | :heavy_check_mark:  |
  | **Class**  | 90  | :heavy_check_mark:  |

**Comment by VovaStelmashchuk** [comment]
Ok. Do i need to add new tests for increase branch test coverage

You can have my help with codecov if your want to.

**Comment by stevenleadbeater** [comment]
Sorry @VovaStelmashchuk I missed that comment. No it's a pre-existing issue. The branch coverage is for the repository as a whole. I'll alter that action to allow different limits for the different areas. That way it can be lowered for branch so we can ensure future work doesn't reduce coverage further but not flag new work as causing a pre-existing issue

**Comment by VovaStelmashchuk** [comment]
Ок.
@jdorn Can we merge the PR ?

**Pull request merged**